src/main/obsRecorder.ts

```diff
@@ -259,13 +264,28 @@ const stop = async () => {
   console.log("[OBS] obsRecorder: stop");
   osn.NodeObs.OBS_service_stopRecording();
 
-  let signalInfo = await waitQueue.shift();
+  // Reject the promise if we don't get a response in time.
```

Comment:
I think these 3 could probably be a `for` loop or something as they are identical, except for the log message
Perhaps:

```typescript
['stopping', 'stop', 'wrote'].forEach(signal => {
  // Reject the promise if we don't get a response in time.
  signalInfo = await Promise.race([
    waitQueue.shift(), 
    new Promise((_r, rej) => setTimeout(rej, 5000))
  ]);

  assertSignal(signalInfo, "recording", signal);
});
```

Reply:
Have simplified this a lot, opted not for a loop in the end.